packages/commonwealth/client/scripts/views/layout.tsx

```diff
@@ -116,15 +117,13 @@ class LayoutComponent extends ClassComponent<LayoutAttrs> {
       this.loadingScope = scope;
       if (scopeMatchesChain) {
         this.deferred = deferChain;
-        selectChain(scopeMatchesChain, deferChain).then((response) => {
-          if (!deferChain && response) {
-            initChain().then(() => {
-              this.redraw();
-            });
-          } else {
-            this.redraw();
-          }
-        });
+        this.loadingChain = true;
+        selectChain(scopeMatchesChain, deferChain)
+          .then((response) => {
+            if (!deferChain && response) initChain().then(() => this.redraw());
+            else this.redraw();
```

Comment:
Please do not remove curly braces, as the readibility of the code is much worse.